### DISCUSSION TITLE
Time for v70?

### DISCUSSION THREAD

**Discussion by abravalheri** [pypa/setuptools#4345] [Dev]
There has been some improvements in the last weeks, so I think we should release something before we start compounding more changes.

The major "breaking changes" are related to removal of deprecated code: https://github.com/pypa/setuptools/pull/4322

I am planning something still this week, beginning of next week max?

**Comment by abravalheri** [comment]
There is another removal scheduled for next week, so we might as well just wait a bit and do it all at once:

https://github.com/pypa/setuptools/blob/22124220568d31d2811f20d1338871af46ce01c6/setuptools/dep_util.py#L5-L16

Update on https://github.com/pypa/setuptools/pull/4360.

**Comment by jaraco** [comment]
Usually, my philosophy is release early and often, as batching changes only complicates triage when there are issues. If we were to release v70 this week and v71 next week, that would increase the churn and bump the number, but it also means that a user with an issue can readily test against those two versions and determine which one (if any) gave them trouble. Maybe they can adopt v70 but not v71. Or maybe neither works, but in either case, they have more information to work with to diagnose and respond.

That said, if #4360 is about to land, I might be tempted to wait, and I have no problem waiting a week if there's nothing pressing that users are waiting on.

**Comment by abravalheri** [comment]
Understood.

I am thinking of the following:

- [x] We merge https://github.com/pypa/setuptools/pull/4360
     - breaking change, but other than that, nothing is pending
- [x] Release v70

Then we can:

- [x] Merge https://github.com/pypa/setuptools/pull/4310
      - I have no problem in merging the PR as it is, but it would be nice if we can solve the pending problems before a new release.
- [ ] Create a follow-up PR addressing the `dynamic` aspects, ensuring PEP 621 is followed
- [ ] Release v70.1
    - I believe the change to `pyproject.toml` does not represent a change in APIs.

**Reply by jaraco** [reply]
#4310 is merged. I seem to think we can live with the dynamic behavior indefinitely (or until someone can prioritize a longer-term fix).
